stac_fastapi/elasticsearch/stac_fastapi/elasticsearch/database_logic.py

```diff
@@ -220,10 +215,7 @@ async def execute_search(
         es_response = await search_task
 
         hits = es_response["hits"]["hits"]
-        items = [
-            self.item_serializer.db_to_stac(hit["_source"], base_url=base_url)
-            for hit in hits
-        ]
+        items = [hit["_source"] for hit in hits]
```

Comment:
Moving db_to_stac introduces an extra loop but I am sure there is no reason to have a loop here to get these results. There must be a better way.

Reply:
```suggestion
        items = (hit["_source"] for hit in hits)
```
Make it a genexp instead of a listcomp and change the return type to Iterable[Dict[str, Any]]

Also, you could probably just make all these signatures more generic and return Iterable instead of List.

Reply:
Like in mongo you could just return a list from your query - do we have to have a loop here to get a list with es?

Reply:
Genexp does have performance advantages - I had never heard of it before

Reply:
> Like in mongo you could just return a list from your query - do we have to have a loop here to get a list with es?

yes, we have to do something like looping, since the item bodies (_source) are an attribute within the list of hits.